lib/widgets/text.dart

```diff
@@ -533,3 +537,162 @@ class _TextWithLinkState extends State<TextWithLink> {
       span);
   }
 }
+
+/// Data to size and position an icon in a span of text.
+class InlineIconGeometryData {
+  /// How tall the icon should be,
+  /// as a fraction of the surrounding text's font size.
+  final double sizeFactor;
+
+  /// Where to assign the icon's baseline, as a fraction of the icon's height,
+  /// when the span is rendered with [TextBaseline.alphabetic].
+  ///
+  /// This is ignored when the span is rendered with [TextBaseline.ideographic];
+  /// zero is used instead.
+  final double alphabeticBaselineFactor;
+
+  /// How much padding should separate the icon from surrounding text,
+  /// as a fraction of the text's font size.
+  final double paddingFactor;
+
+  const InlineIconGeometryData._({
+    required this.sizeFactor,
+    required this.alphabeticBaselineFactor,
+    required this.paddingFactor,
```

Comment:
It looks like the baseline factor ends up getting interpreted compounded with the size factor, but the padding factor doesn't.

Is there a motivation for that difference? Seems like it'd be easy to get confused about. It might be cleanest to specify both of them the same way (either both relative to the surrounding font size, or both relative to the icon size).